cmds/webboot/webboot_test.go

```diff
@@ -44,64 +43,76 @@ func TestDownload(t *testing.T) {
 	})
 }
 
-func TestDownloadByLinkOption(t *testing.T) {
-	url := "http://tinycorelinux.net/10.x/x86_64/release/TinyCorePure64-10.1.iso"
-	name := "test_download_by_link.iso"
-	fPath := "/tmp/test_download_by_link.iso"
-
-	uiEvents := make(chan ui.Event)
-	input := append(strings.Split(url, ""), "<Enter>")
-	input = append(input, strings.Split(name, "")...)
-	input = append(input, "<Enter>")
-	go pressKey(uiEvents, input)
-	var entry menu.Entry = &DownloadByLink{uiEvents: uiEvents}
-
-	if err := entry.Exec(); err != nil {
-		t.Errorf("Fail to execute, error msg: %+v", err)
-	}
-
-	if _, err := os.Stat(fPath); err != nil {
-		t.Errorf("Fail to find downloaded file, error msg: %+v", err)
-	}
-
-	if err := os.Remove(fPath); err != nil {
-		t.Errorf("Fail to remove test file, error msg: %+v", err)
+func TestDownloadOption(t *testing.T) {
+	bookmarkIso := &ISO{
+		label: "TinyCorePure64-10.1.iso",
+		path:  "/tmp/TinyCorePure64-10.1.iso",
 	}
-}
-
-func TestBookmarkISOOption(t *testing.T) {
-	var entry menu.Entry = bookmark[0]
-	fPath := "/tmp/TinyCorePure64-10.1.iso"
-
-	if err := entry.Exec(); err != nil {
-		t.Errorf("Fail to execute, error msg: %+v", err)
+	downloadByLinkIso := &ISO{
+		label: "test_download_by_link.iso",
+		path:  "/tmp/test_download_by_link.iso",
 	}
-
-	if _, err := os.Stat(fPath); err != nil {
-		t.Errorf("Fail to  find downloaded file, error msg: %+v", err)
+	downloadLink := "http://tinycorelinux.net/10.x/x86_64/release/TinyCorePure64-10.1.iso"
+
+	for _, tt := range []struct {
+		name  string
+		label []string
+		url   []string
+		want  *ISO
+	}{
+		{
+			name:  "test_bookmark",
+			label: append(strings.Split(bookmarkIso.label, ""), "<Enter>"),
+			want:  bookmarkIso,
+		},
+		{
+			name:  "test_download_by_link",
+			label: append(strings.Split(downloadByLinkIso.label, ""), "<Enter>"),
+			url:   append(strings.Split(downloadLink, ""), "<Enter>"),
+			want:  downloadByLinkIso,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			uiEvents := make(chan ui.Event)
+			input := append(tt.label, tt.url...)
+			go pressKey(uiEvents, input)
+
+			downloadOption := DownloadOption{}
+			entry, err := downloadOption.exec(uiEvents)
+
+			if err != nil {
+				t.Errorf("Error: %v", err)
+			}
+			iso, ok := entry.(*ISO)
+			if !ok {
+				t.Errorf("Fail to transfor the entry to *ISO, error msg: %+v", err)
+			}
+			if tt.want.label != iso.label || tt.want.path != iso.path {
+				t.Errorf("Incorrect return. get %+v, want %+v", entry, tt.want)
+			}
+			if _, err := os.Stat(iso.path); err != nil {
+				t.Errorf("Fail to find downloaded file, error msg: %+v", err)
+			}
+
+			if err := os.Remove(iso.path); err != nil {
+				t.Errorf("Fail to remove test file, error msg: %+v", err)
```

Comment:
For future reference, I think its ok if you dont say "error msg:", you can just put a colon after your message and print the err